executor.py

```diff
@@ -1,7 +1,114 @@
-from jina import Executor, DocumentArray, requests
+from jina import Executor, requests
+from typing import Optional, Dict, Any, List
+from docarray import DocumentArray
+from jina.logging.logger import JinaLogger
 
 
 class ElasticSearchIndexer(Executor):
-    @requests
-    def foo(self, docs: DocumentArray, **kwargs):
-        pass
+    def __init__(
+        self,
+        hosts: Optional[str] = 'http://localhost:9200',
+        n_dim: Optional[int] = 128,
+        distance: Optional[str] = 'cosine',
+        index_name: Optional[str] = None,
+        es_config: Optional[Dict[str, Any]] = None,
+        index_text: Optional[bool] = False,
+        tag_indices: Optional[List[str]] = None,
+        batch_size: Optional[int] = 64,
+        ef_construction: Optional[int] = None,
+        m: Optional[int] = None,
+        **kwargs,
+    ):
+
+        super().__init__(**kwargs)
+
+        self._index = DocumentArray(
+            storage='elasticsearch',
+            config={
+                'hosts': hosts,
+                'n_dim': n_dim,
+                'distance': distance,
+                'index_name': index_name,
+                'es_config': es_config or {},
+                'index_text': index_text,
+                'tag_indices': tag_indices or [],
+                'batch_size': batch_size,
+                'ef_construction': ef_construction,
+                'm': m
+            },
+        )
+
+        self.logger = JinaLogger(self.metas.name)
+
+    @requests(on='/index')
+    def index(self, docs: DocumentArray, **kwargs):
+
+        if docs:
```

Comment:
is this `if` needed?

Reply:
DocumentArray should be able to hanlde `extend` from an empty docarray transparently

Reply:
done